flexget/utils/requests.py

```diff
@@ -145,13 +145,23 @@ def request(self, method, url, *args, **kwargs):
 
         # If we do not have an adapter for this url, pass it off to urllib
         if not any(url.startswith(adapter) for adapter in self.adapters):
+            log.debug('No adaptor, passing off to urllib')
             return _wrap_urlopen(url, timeout=kwargs['timeout'])
 
         try:
+            log.debug('Fetching %s' % url)
             result = requests.Session.request(self, method, url, *args, **kwargs)
-        except (requests.Timeout, requests.ConnectionError):
-            # Mark this site in known unresponsive list
-            set_unresponsive(url)
+        except (requests.Timeout, requests.ConnectionError) as e:
+            msg = str(e)
+            log.debug('Exception fetching %s: %s' % (url, msg))
+            # torcache.net answers HTTP GET requests but doesn't properly answer HTTP HEAD requests - it outputs an
+            # empty string which causes a ProtocolError to be raised, which eventually turns into a ConnectionError
+            # exception.  urlrewriting unconditionally does a HEAD request for all URLs; this results in torcache.net
+            # being marked as unresponsive if you rewrite its URLs.  Don't add a site to the known unresponsive list
+            # when the exception message indicates it was a protocol error.
+            if 'ProtocolError' not in msg or 'Connection aborted' not in msg:
+                # Mark this site in known unresponsive list
+                set_unresponsive(url)
```

Comment:
I think the reason connectionerror was even in there is because requests didn't wrap all timeouts in the timeout exception before. I believe this might be fixed on newer requests versions though, perhaps we should just remove that exception.

(this was the original timeout that came through as a ConnectionError https://gist.github.com/Gargauth/b54298ca62119836ddec#file-flexget-log-L85)